autoload/maktaba/plugin.vim

```diff
@@ -102,6 +111,30 @@ function! s:SanitizedName(name) abort
 endfunction
 
 
+""
+" Gets a dictionary of {leaf path} for each path found on &rtp that does not
+" correspond to a registered plugin.
+function! s:GetUnregisteredLeafdirs() abort
+  let l:cache = s:unregistered_leafdirs_cache
+  let l:plugin_locations = keys(s:plugins_by_location)
+  if !has_key(l:cache, 'rtp') || !has_key(l:cache, 'plugin_locations') ||
+      \ l:cache.rtp isnot# &rtp ||
```

Comment:
The difference is that `==` will do insane type conversions if you accidentally compare different types whereas `is` will fail for different types. I lean towards `is` because `==` has such error-prone behavior. Here it doesn't make much difference besides being defensive about the type of `l:cache.rtp`, but I'd rather keep it as-is if it's all the same.